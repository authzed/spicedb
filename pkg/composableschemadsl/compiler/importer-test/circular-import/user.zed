import "subjects.zed"

definition user {}
