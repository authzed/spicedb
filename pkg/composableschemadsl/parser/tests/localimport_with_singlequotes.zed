import 'path/to/user.zed'

definition resource {
    relation user: user
    relation persona: persona
    permission view = user + persona
}
