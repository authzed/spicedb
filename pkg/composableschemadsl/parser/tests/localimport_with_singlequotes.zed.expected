NodeTypeFile
  end-rune = 141
  input-source = local imports with singlequotes on import test
  start-rune = 0
  child-node =>
    NodeTypeImport
      end-rune = 24
      import-path = path/to/user.zed
      input-source = local imports with singlequotes on import test
      start-rune = 0
    NodeTypeDefinition
      definition-name = resource
      end-rune = 140
      input-source = local imports with singlequotes on import test
      start-rune = 27
      child-node =>
        NodeTypeRelation
          end-rune = 71
          input-source = local imports with singlequotes on import test
          relation-name = user
          start-rune = 53
          allowed-types =>
            NodeTypeTypeReference
              end-rune = 71
              input-source = local imports with singlequotes on import test
              start-rune = 68
              type-ref-type =>
                NodeTypeSpecificTypeReference
                  end-rune = 71
                  input-source = local imports with singlequotes on import test
                  start-rune = 68
                  type-name = user
        NodeTypeRelation
          end-rune = 101
          input-source = local imports with singlequotes on import test
          relation-name = persona
          start-rune = 77
          allowed-types =>
            NodeTypeTypeReference
              end-rune = 101
              input-source = local imports with singlequotes on import test
              start-rune = 95
              type-ref-type =>
                NodeTypeSpecificTypeReference
                  end-rune = 101
                  input-source = local imports with singlequotes on import test
                  start-rune = 95
                  type-name = persona
        NodeTypePermission
          end-rune = 138
          input-source = local imports with singlequotes on import test
          relation-name = view
          start-rune = 107
          compute-expression =>
            NodeTypeUnionExpression
              end-rune = 138
              input-source = local imports with singlequotes on import test
              start-rune = 125
              left-expr =>
                NodeTypeIdentifier
                  end-rune = 128
                  identifier-value = user
                  input-source = local imports with singlequotes on import test
                  start-rune = 125
              right-expr =>
                NodeTypeIdentifier
                  end-rune = 138
                  identifier-value = persona
                  input-source = local imports with singlequotes on import test
                  start-rune = 132